plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -103,11 +162,14 @@ public void setSecrets(Map<String, String> secrets) {
         this.secrets = secrets;
     }
 
-    private String getVariable(String key, Properties prop) {
-        if (prop != null && !prop.isEmpty()) {
-            return prop.getProperty(key, System.getenv(key));
-        } else {
-            return System.getenv(key);
-        }
+    private Optional<String> getVariable(String key, Properties prop) {
+        return prop.containsKey(key) ?
```

Comment:
This could just be
```
return Optional.ofNullable(prop.getProperty(key, System.getenv(key)));
```

Reply:
Yes that looks way better. Thank you! :)